core/modules/installer/tests/installer_test/installer_test.module

```diff
@@ -0,0 +1,257 @@
+<?php
+
+
+/**
+ * Implements hook_archiver_info().
+ */
+function installer_test_archiver_info() {
+  return array(
+    'installer_test_archiver' => array(
+      // This is bogus, we only care about the extensions for now.
+      'class' => 'ArchiverInstallerTest',
+      'extensions' => array('installer-test-extension'),
+    ),
+  );
+}
+
+/**
+ * Implements hook_menu().
+ */
+function installer_test_menu() {
+  $items['core/modules/installer/tests/installer_test_xml/%/%'] = array(
+    'title' => 'XML feed page',
+    'description' => "Deliver release info as an XML file.",
+    'page callback' => 'installer_test_xml',
+    'page arguments' => array(5, 6),
+    'access arguments' => array('access content'),
+    'type' => MENU_CALLBACK,
+  );
+  $items['installer_test/query/%/%'] = array(
+    'title' => 'Test Query page',
+    'description' => "Tests the ability to fetch and display projects, and filter them appropriately.",
+    'page callback' => 'installer_test_query',
+    'page arguments' => array(1, 2, 3),
+    'access arguments' => array('access content'),
+    'type' => MENU_CALLBACK,
+  );
+  return $items;
+}
+
+/**
+ * Page callback. Outputs release information for test projects as XML data.
+ */
+function installer_test_xml($project = NULL, $version = '1.x') {
+  $path = backdrop_get_path('module', 'installer'); 
+  readfile("$path/tests/installer_test_xml/$project.xml");
+  backdrop_exit();
+}
+
+/**
+ * Page callback. Generates json based on the input filters
```

Comment:
This and most functions in this file could use a trailing period on all function descriptions.